client/server/pages/index.js

```diff
@@ -565,15 +565,6 @@ export default function pages() {
 	app.use( setupLoggedInContext );
 	app.use( handleLocaleSubdomains );
 
-	// redirect homepage if the Reader is disabled
-	app.get( '/', function ( request, response, next ) {
```

Comment:
Removing this, we'll also have to remove a test: https://github.com/Automattic/wp-calypso/blob/master/client/server/pages/test/index.js#L1359